### PR TITLE
Camera init cinematic

### LINKED ISSUE #969: [REFACTOR] Improve battle initialization
As soon as the match starts (hence the scene loads), the camera is not fixed on the player's character and pans to it.
This must be set before showing anything to the player.

### PR DESCRIPTION
**Pull request by BertovDev**
This PR add a cinematic introduction to the battle Scene, includes:

- A loading screen that helps to hide the Init phase of the Map, players and camera
- A cinematic that shows part of the map and finish on your character
- A UI splash with an animation and text.

Relatated issues:

#989 
#988 

closes #969 

Captura de pantalla 2023-10-18 a la(s) 11 59 52

- [x] Tested in Unity Desktop
- [x] Tested in Android

### CODE REVIEW

**Review by agustinesco — CHANGES_REQUESTED** [review]
Testing the pr i found out that in some screens the message doesn't fully cover the screen in particular with the "motorola moto E" simulator:
image

Also when we reconnect to a game we got the same message with the same camera movement. Maybe we could change the message or omit the camera movement on the reconnect escenarios, what do you think?

File: client/Assets/Scripts/CustomLevelManager.cs [hunk 1]
```diff
@@ -52,6 +53,18 @@ public class CustomLevelManager : LevelManager
     public List<CoMCharacter> charactersInfo = new List<CoMCharacter>();
     public List<GameObject> mapList = new List<GameObject>();
 
+    //Camera cinematic variables
+    [SerializeField]
+    GameObject loadinScreen;
```
Comment (ncontinanza) [inline client/Assets/Scripts/CustomLevelManager.cs:58]:
```suggestion
    GameObject loadingScreen;
```

File: client/Assets/Scripts/CustomLevelManager.cs [hunk 2]
```diff
@@ -178,6 +202,67 @@ private void GeneratePlayers()
...
     }
 
+    IEnumerator CameraCinematic()
+    {
+        if (!SocketConnectionManager.Instance.cinematicDone)
+        {
+            //Start moving camera and remove loading sceen
+            InvokeRepeating("Substract", 1f, 0.1f);
+            yield return new WaitForSeconds(1.7f);
+            loadinScreen.SetActive(false);
```
Comment (ncontinanza) [inline client/Assets/Scripts/CustomLevelManager.cs:212]:
```suggestion
            loadingScreen.SetActive(false);
```

File: client/Assets/Scripts/CustomLevelManager.cs [hunk 3]
```diff
@@ -178,6 +202,67 @@ private void GeneratePlayers()
...
+            InvokeRepeating("MoveYCamera", 0.3f, 0.1f);
+            //Cancel camera zoom
+            yield return new WaitForSeconds(0.5f);
+            CancelInvoke("MoveYCamera");
+        }
+        else
+        {
+            cameraFramingTransposer.m_TrackedObjectOffset = new Vector3(0, 0, 0);
+            yield return new WaitForSeconds(0.8f);
+            loadinScreen.SetActive(false);
```
Comment (ncontinanza) [inline client/Assets/Scripts/CustomLevelManager.cs:229]:
```suggestion
            loadingScreen.SetActive(false);
```

File: client/Assets/Scripts/CustomLevelManager.cs [hunk 4]
```diff
@@ -178,6 +202,67 @@ private void GeneratePlayers()
...
+        }
+    }
+
+    int roundupbyten(int i)
```
Comment (ncontinanza) [inline client/Assets/Scripts/CustomLevelManager.cs:233]:
```suggestion
    int roundUpByTen(int i)
```

File: client/Assets/Scenes/Battle.unity [hunk 5]
```diff
@@ -443,16 +553,26 @@ PrefabInstance:
...
       value: 
       objectReference: {fileID: 2856279044554654234}
+    - target: {fileID: 8472880968645199034, guid: 62eb30154c76d4e0fb902ff01db61581,
+        type: 3}
+      propertyPath: battleScreen
+      value: 
+      objectReference: {fileID: 2856279044554654232}
+    - target: {fileID: 8472880968645199034, guid: 62eb30154c76d4e0fb902ff01db61581,
+        type: 3}
+      propertyPath: loadinScreen
```
Comment (ncontinanza) [inline client/Assets/Scenes/Battle.unity:563]: This should be changed to fix the typo as well

**Review by ncontinanza — CHANGES_REQUESTED** [review]

**Review by agustinesco — APPROVED** [review]
LGTM!

**Review by agustinesco — DISMISSED** [review]

**Review by ncontinanza — DISMISSED** [review]
LGTM!

**Review by agustinesco — APPROVED** [review]
LGTM!

**Review by ncontinanza — APPROVED** [review]

**Pull request merged**